src/sidebar/oauth-auth.js

```diff
@@ -79,14 +80,20 @@ function auth($http, settings) {
 
   // Set a timeout to refresh the access token a few minutes before it expires.
   function refreshAccessTokenBeforeItExpires(tokenInfo) {
-    var delay = tokenInfo.expiresIn * 1000;
-
     // We actually have to refresh the access token _before_ it expires.
     // If the access token expires in one hour, this should refresh it in
     // about 55 mins.
-    delay = Math.floor(delay * 0.91);
+    var refreshAfter = tokenInfo.expiresAt.valueOf() * 0.91;
+
+    function refreshAccessTokenIfNearExpiry() {
+      if (Date.now() > refreshAfter) {
+        refreshAccessToken(tokenInfo.refreshToken);
```

Comment:
`refreshAccessToken() calls this function (`refreshAccessTokenBeforeItExpires()`) again with the new access token, after it is successfully received from the server. If the refresh request fails then `refreshAccessToken()` does not call this function again, and no further refresh requests will be sent after one fails. (This is the same as the existing behaviour of this code.)